python_ta/checkers/redundant_assignment_checker.py

```diff
@@ -0,0 +1,132 @@
+"""Checker for redundant assignment statements in the program.
+
+An assignment statement is redundant if it satisfies the following two properties:
+
+    1. Every path from a definition of a variable `v` to its first usage
+    goes through at least one re-definition of `v`.
+
+    2. Removing the statement from the program does not in any way change
+    the behavior of the program.
+"""
+from typing import Union
+import astroid
+from astroid.node_classes import NodeNG
+from pylint.interfaces import IAstroidChecker
+from pylint.checkers import BaseChecker, utils
+from pylint.checkers.utils import check_messages
+from python_ta.cfg.graph import CFGBlock, ControlFlowGraph
+from typing import Set, List
+
+
+class RedundantAssignmentChecker(BaseChecker):
+
+    __implements__ = IAstroidChecker
+    # name is the same as file name but without _checker part
+    name = 'redundant_assignment'
+    # use dashes for connecting words in message symbol
+    msgs = {'E9959': ('This assignment statement is redundant;' \
+                      ' You can remove it from the program.',
+                      'redundant-assignment',
+                      'This assignment statement is redundant;' \
+                      ' You can remove it from the program.')
+           }
+
+    # this is important so that your checker is executed before others
+    priority = -1
+
+    def __init__(self, linter=None):
+        super().__init__(linter=linter)
+        self._redundant_assignment: Set[astroid.Assign] = set()
+
+    @check_messages('redundant-assignment')
+    def visit_assign(self, node: astroid.Assign):
+        if node in self._redundant_assignment:
+            self.add_message('redundant-assignment', node=node)
+
+    def visit_module(self, node: astroid.Module):
+        self._analyze(node)
+
+    def visit_functiondef(self, node: astroid.FunctionDef):
+        self._analyze(node)
+
+    def _analyze(self, node: Union[astroid.Module, astroid.FunctionDef]) -> None:
+        """Runs the backward data flow algorithm on a `Module` or `Function` CFG, which in turn
+        appends `Assign` nodes to `self.redundant_assignment` if it is redundant.
+
+        Data flow algorithms retrieved from:
+        https://www.seas.harvard.edu/courses/cs252/2011sp/slides/Lec02-Dataflow.pdf#page=31
+        """
+        # stores all the variables that will be re-defined before any usage at a
```

Comment:
Capitalize "stores". Also your documentation here should specify the type better; you can say the "variable names" (which implies strings) rather than just "variables" (which might be a kind of astroid node).